### ISSUE TITLE
Clang gives false warning when using designated initializer with brace intialisation

### ISSUE THREAD

**Issue by ranaanoop** [llvm/llvm-project#171790]
The following well-formed program produces a warning: [demo](https://godbolt.org/z/cKGbPE9Wv)

```
struct S {
    int x;
};

S s{
    .x{0}  // Warning: Braces around scalar initializer
};
```
The warning says:

```
<source>:6:7: warning: braces around scalar initializer [-Wbraced-scalar-init]
    6 |     .x{0}  // Warning: Braces around scalar initializer
      |       ^~~
      |    
```

**Comment by shafik** [comment]
It is a diagnotic warning showing suspicious but not neccessirily ill-formed code, looks like this was 420fa12dfddc3 and it looks intentional.

CC @zygoloid

**Comment by MagentaTreehouse** [comment]
Duplicate of https://github.com/llvm/llvm-project/issues/57286. CC @shafik
